compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -259,7 +258,7 @@ impl Context {
 
         // If the instruction doesn't have side-effects, cache the results so we can reuse them if
         // the same instruction appears again later in the block.
```

Comment:
This comment is a bit incorrect as it's just referring to "side effects" whereas the condition is more specific.